src/scripts/main.js

```diff
@@ -1,11 +1,31 @@
+/* eslint-disable prettier/prettier */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const promise = new Promise((resolve) => {
+    element.addEventListener(eventName, (e) => {
+      // eslint-disable-next-line no-undef
+      resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+    },
+    { once: true });
+  });
+
+  return promise;
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const existingMessage = document.querySelector('.message');
+
+  if (existingMessage) {
+    existingMessage.textContent = message;
+  } else {
+    const div = document.createElement('div');
+
+    div.classList.add('message');
+    div.textContent = message;
+
+    document.querySelector('form').append(div);
```

Comment:
The 'form' selector is too generic. If there are multiple form elements on the page, this will select the first one, which may not be the one you intended. Consider using a more specific selector.